device_backends/NumericAddressedBackend/src/NumericAddressedBackend.cc

```diff
@@ -88,19 +85,27 @@ namespace ChimeraTK {
 
   /* Call 32-bit address implementation by default, for backends that don't implement 64-bit */
   void NumericAddressedBackend::read(uint64_t bar, uint64_t address, int32_t* data, size_t sizeInBytes) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
     read(static_cast<uint8_t>(bar), static_cast<uint32_t>(address), data, sizeInBytes);
+#pragma GCC diagnostic pop
   }
 
   /********************************************************************************************************************/
 
   void NumericAddressedBackend::write(uint64_t bar, uint64_t address, int32_t const* data, size_t sizeInBytes) {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
I think this is the only place where the 32-bit write is being called at all. Not sure whether the backend programmer will see that he/she is overriding a deprecated function. It might be that this disables all the deprecation warnings which have just been introduced.

However, I agree that this needed and correct here. Was just a remark.

Reply:
You are right, the `[[deprecated]]` specifier does not help for overrides. The ignore pragma also does not hurt, since it only affects that single line (until the `#pragma GCC diagnostic pop`).